packages/core/src/model/TransactionOptions.ts

```diff
@@ -6,4 +6,9 @@ import { Signer } from 'ethers'
 export interface TransactionOptions {
   signer?: Signer
   transactionName?: string
+  privateKey?: string
```

Comment:
Does signer and privateKey fields make sense being placed in the same object? If not maybe something like this:
```ts
export interface TransactionOptionBase {
  transactionName?: string
}
export interface TransactionOptionWithSigner {
  signer?: Signer
}
export interface TransactionOptionsWithPrivateKey {
  privateKey?: string,
  chainId?: number
}
...
export type TransactionOptions = TransactionOptionBase & (TransactionOptionWithSigner | TransactionOptionsWithPrivateKey | ...)
```